app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/image_import_workflow.rb

```diff
@@ -42,13 +42,19 @@ def post_execute
 
     ems = ExtManagementSystem.find(options[:ems_id])
 
+    cos_pvs_creds = options[:cos_pvs_creds]
+    bearer = Authentication.find(cos_pvs_creds[:bearer_id])
```

Comment:
What happens if it's not found?  This will blow up I think, which is probably good.

What happens if somebody passes a random id here...can they then extract authentications from other providers?

At a minimum it feels like we should be scoping the authentications by ems, so I'd expect something like `ems.authentications.find(...)`, but even so, we should probably only look for specific kinds of authentications, to keep the exposure to the minimum.

Reply:
Good catch. Spoofing COS id will send COS creds to some PowerVC machine. E.g. one register's a PowerVC provider and chooses COS owned by another user. Then COS creds of that user will be sent to the PowerVC machine. One way to fix that would be associating COS with a given PowerVC provider during provider registration. Re-association should be allowed only if COS creds are revalidated.

Reply:
> One way to fix that would be associating COS with a given PowerVC provider during provider registration.

I like this idea - during COS registration select PowerVC instances that are allowed to access the credentials.